player/engine.py

```diff
@@ -112,13 +131,17 @@ def bot_engine(self):
             if game_state.current_turn == game_state.num_turns:
                 break
 
-            # move each one of vehicles
-            for vehicle_id, vehicle in self._allyVehicles.items():
-                # try to shot
-                shooted = self._shoot_with_vehicle(vehicle_id, vehicle)
+            spg = {vehicle_id: vehicle for vehicle_id, vehicle in self._allyVehicles.items() if vehicle.vehicle_type == VehicleType.SPG}
+            light = {vehicle_id: vehicle for vehicle_id, vehicle in self._allyVehicles.items() if vehicle.vehicle_type == VehicleType.LIGHT_TANK}
+            heavy = {vehicle_id: vehicle for vehicle_id, vehicle in self._allyVehicles.items() if vehicle.vehicle_type == VehicleType.HEAVY_TANK}
+            medium = {vehicle_id: vehicle for vehicle_id, vehicle in self._allyVehicles.items() if vehicle.vehicle_type == VehicleType.MEDIUM_TANK}
+            at_spg = {vehicle_id: vehicle for vehicle_id, vehicle in self._allyVehicles.items() if vehicle.vehicle_type == VehicleType.AT_SPG}
 
-                # now move to the center if didn't shot
-                if shooted == False:
-                    self.__execute_movement(vehicle_id, vehicle)
+            # move each one of vehicles
+            self.__vehicle_action(spg)
+            self.__vehicle_action(light)
+            self.__vehicle_action(heavy)
+            self.__vehicle_action(medium)
+            self.__vehicle_action(at_spg)
```

Comment:
Why filter vehicles by type if they are passed to the same function anyway?

Reply:
Because we need exact orders of vehicles. Also in future there can be more than one vehicle of same type.

Reply:
Because we can move vehicles in exact order. Also in future there can be more than one vehicle of the same type.

Reply:
Is not this code supports more than one vehicle of each type right now? Also what is the benefit to move vehicles in particular order?

Reply:
It's a rule to move in exact order on server you can see it in documentation.